src/node_io.h

```diff
@@ -0,0 +1,236 @@
+#ifndef SRC_NODE_IO_H_
+#define SRC_NODE_IO_H_
+
+#include "node.h"
+
+namespace node {
+
+namespace io {
+
+enum io_error {
+  // Source would overflow the provided buffer
+  IO_ERROR_OVERFLOW = -256,
+  // Source does not support FLAG_PEEK
+  IO_ERROR_NO_PEEK = -512,
+  // Attempt to pull from an already ended source
+  IO_ERROR_EOF = -768,
+  // Source will not copy data into caller provided buffer
+  IO_ERROR_MUST_NO_COPY = -1024,
+};
```

Comment:
Are errors distinguished any further than this? I think you’d need at least a way to forward all `errno` errors…

Reply:
So far, not yet. This is just a rough error reporting mechanism right now, it would need to be significantly improved :-)